src/renderer/port-forward/port-forward-store/port-forward-store.ts

```diff
@@ -322,7 +325,7 @@ export class PortForwardStore extends ItemStore<PortForwardItem> {
 
     try {
       // check if the port-forward is active, and if so check if it has the same local port
-      const pf = await getActivePortForward(portForward);
+      const pf = await this.dependencies.requestActivePortForward(portForward);
 
       if (pf?.forwardPort && pf.forwardPort !== portForward.forwardPort) {
```

Comment:
Hmm. `pf` and `portForward` are the same reference, and therefore `pf.forwardPort !== portForward.forwardPort` is always `false`.

Reply:
No they are no, `pf` is the return value of `requestActivePortForward`, they might be the same type but they aren't the same reference.

Reply:
But I completely agree with you that we should never modify the inputs as that is very surprising.